src/config/settings.ts

```diff
@@ -212,7 +232,11 @@ export function updateFromCommandLine(cfg: Settings, args: string[]): Settings {
     i += 1;
   }
   if (lim > 0) {
+    if (cfg.noModalDialogs) {
+      cfg.kubernetes.suppressSudo = true;
```

Comment:
That's what I meant by "transient option" in https://github.com/rancher-sandbox/rancher-desktop/pull/2039#issuecomment-1099785619:

I expect (after the engine refactoring) that each `set` API call will create some kind of context that will be passed along internally until the desired state has been reached. The transient options ("no sudo", "no modal dialog") would override whatever permanent settings we have, but for this transaction only.

I understand that this may not be possible until after the engine refactoring.

Reply:
Yup, I wanted to avoid changing the interface of LimaBackend / K8s.KubernetesBackend but this would require such a transient setting.

Reply:
And the new PR implements a TransientSettings type of exactly one property.